src/gretel_synthetics/generator.py

```diff
@@ -0,0 +1,236 @@
+from typing import TYPE_CHECKING, Callable, List, Iterable, Optional, Tuple
+from dataclasses import dataclass, asdict
+from collections import namedtuple
+
+import cloudpickle
+
+import sentencepiece as spm
+import tensorflow as tf
+
+from .model import _build_sequential_model
+
+if TYPE_CHECKING:
+    from .config import BaseConfig, LocalConfig
```

Comment:
Please use full package names here. 

Also, in other packages, we've been doing:

```python
if TYPE_CHECKING:
    from foo import Bar
else:
    Bar = None
```

This let's us still used the reference as a type hint and not have to use a forward reference string.